services/background-worker/src/Wire/BackendDeadUserNotificationWatcher.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE BlockArguments #-}
+
+module Wire.BackendDeadUserNotificationWatcher where
+
+import Cassandra
+import Control.Error.Util (hush)
+import Control.Monad.Trans.Maybe
+import Data.ByteString.Conversion
+import Data.Domain
+import Data.Id
+import Data.Text qualified as Text
+import Data.Text.Encoding qualified as Text
+import Imports
+import Network.AMQP qualified as Q
+import Network.AMQP.Extended
+import Network.AMQP.Lifted qualified as QL
+import UnliftIO
+import Wire.BackgroundWorker.Env
+
+type DomainConsumerMap = IORef (Map Domain (Q.ConsumerTag, MVar ()))
+
+startWatcher :: DomainConsumerMap -> Q.Channel -> AppT IO ()
+startWatcher _consumersRef chan = do
+  markAsWorking BackendDeadUserNoticationWatcher
+
+  cassandra <- asks (.cassandra)
+
+  -- This ensures that we receive notifications 1 by 1 which ensures they are
+  -- delivered in order.
+  lift $ Q.qos chan 0 1 False
```

Comment:
Why is the order important here? This would definitely affect the throughput.

Reply:
Leftover from copying from the other bg worker. Gone now.